crates/bevy_ecs/src/archetype.rs

```diff
@@ -33,16 +34,22 @@ impl ArchetypeId {
     }
 }
 
+/// Records whether a component was modified by addition or mutation
```

Comment:
(Specifically in the case of adding a bundle to an archetype, where adding means that it was not in the archetype, and mutating means it was)